### ISSUE TITLE
why train loss only record the latest batch?

### ISSUE THREAD

**Issue by murphypei** [OlgaChernytska/2D-Hand-Pose-Estimation-RGB#2]
Hi, thanks for your job. I'm a little confused that the running_loss is only computed by the last batch but not all batches in each epoch, is it reasonable?

```python
    def _epoch_train(self, dataloader):
        self.model.train()
        running_loss = []

        for i, data in enumerate(dataloader, 0):
            inputs = data["image"].to(self.device)
            labels = data["heatmaps"].to(self.device)

            self.optimizer.zero_grad()
            running_loss = []

            outputs = self.model(inputs)
            loss = self.criterion(outputs, labels)
            loss.backward()
            self.optimizer.step()

            running_loss.append(loss.item())

            if i == self.batches_per_epoch:
                epoch_loss = np.mean(running_loss)
                self.loss["train"].append(epoch_loss)
                break
```

**Comment by OlgaChernytska** [comment]
Oh, you are right. Thank you!
I'll fix it now.

**Comment by OlgaChernytska** [comment]
Looks like this error does not affect the training so much.
It does not affect how the optimizing step is performed. It affects only the ReduceLROnPlateau loss scheduler.

**Comment by murphypei** [comment]
> d. It affects only the ReduceLROnPlateau loss sche

yeah, thank you for your reply.
